Readability.js

```diff
@@ -2104,14 +2117,28 @@ Readability.prototype = {
   },
 
   /**
-   * Clean out spurious headers from an Element. Checks things like classnames and link density.
+   * Clean out spurious headers from an Element.
    *
    * @param Element
    * @return void
   **/
   _cleanHeaders: function(e) {
-    this._removeNodes(this._getAllNodesWithTag(e, ["h1", "h2"]), function (header) {
-      return this._getClassWeight(header) < 0;
+    this._removeNodes(this._getAllNodesWithTag(e, ["h1", "h2"]), function (node) {
+      // remove heading if it is first element
+      let prev = node.previousSibling || (node.parentNode || {}).previousSibling;
+      while (prev && this._isWhitespace(prev)) {
+        prev = prev.previousSibling || (prev.parentNode || {}).previousSibling;
+      }
+      if (!prev) {
+        return true;
+      }
```

Comment:
Hm, did we do this before? I'm worried about this especially because some high-profile sites (CNN, if I remember correctly) like to use `h2` for their introductory paragraphs, and obviously we would be better off not cutting those entirely...

Reply:
it wasn't specified like this but practically we did. (because 19 test change when we don't do it). I will remove this block as you mentioned above that you want to keep those headers.